README.md

```diff
@@ -0,0 +1,335 @@
+	err = client.Connect(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer client.Disconnect(ctx)
+	database := client.Database("alexa")
+	connection := Connection{
+		database: database,
+	}
+	lambda.Start(connection.IntentDispatcher)
+}
+```
+
+Notice in the above code that we've added a `lambda.Start` call in our `main` function that points to an `IntentDispatcher` function. We're designing this function to use the connection information established in the `main` function, which based on our Lambda knowledge, may not run every time the function is executed.
+
+So we've got the foundation to our Alexa Skill in place. Now we need to design the logic for each of our intents that were previously defined in the Alexa Developer Portal.
+
+Since this is going to be a recipe related Skill, let's model our MongoDB documents like the following:
+
+```json
+{
+    "_id": ObjectID("234232358943"),
+    "name": "chocolate chip cookies",
+    "ingredients": [
+        "flour",
+        "egg",
+        "sugar",
+        "chocolate"
+    ]
+}
+```
+
+There is no doubt that our documents could be more extravagant, but for this example it will work out fine. Within the MongoDB Atlas cluster, create the **alexa** database if it doesn't already exist and add a document modeled like the above in a **recipes** collection.
+
+In the `main.go` file of the project, add the following data structure:
+
+```go
+type Recipe struct {
+	Id          primitive.ObjectID `bson:"_id"`
+	Name        string             `bson:"name"`
+	Ingredients []string           `bson:"ingredients"`
+}
+```
+
+With the MongoDB Go driver, we can annotate Go data structures with BSON so that way we can easily map between the two. It essentially makes our lives a lot easier when working with MongoDB and Go.
+
+Let's circle back to the `IntentDispatcher` function:
+
+```go
+func (connection Connection) IntentDispatcher(ctx context.Context, request alexa.Request) (alexa.Response, error) {
+	var response alexa.Response
+	switch request.Body.Intent.Name {
+	case "GetIngredientsForRecipeIntent":
+	case "GetRecipeFromIngredientsIntent":
+	default:
+		response = alexa.NewSimpleResponse("Unknown Request", "The intent was unrecognized")
+	}
+	return response, nil
+}
+```
+
+Remember the two intents from the Alexa Developer Portal? We need to assign logic to them.
+
+Essentially, we're going to do some database logic and then use the `NewSimpleResponse` function to create a response the the results.
+
+Let's start with the `GetIngredientsForRecipeIntent` logic:
+
+```go
+case "GetIngredientsForRecipeIntent":
+    var recipe Recipe
+    recipesCollection := connection.database.Collection("recipes")
+    recipeName := request.Body.Intent.Slots["recipe"].Value
+    if recipeName == "" {
+        return alexa.Response{}, errors.New("Recipe name is not present in the request")
+    }
+    if err := recipesCollection.FindOne(ctx, bson.M{"name": recipeName}).Decode(&recipe); err != nil {
```

Comment:
How do errors behave for Alexa skills? For example, if a recipe isn't found, `FindOne` will return `mongo.ErrNoDocuments`. Should that be reported as an error here or as a `alexa.NewSimpleResponse`?

Reply:
The Lambda SDK should properly process the error.